### ISSUE TITLE
middleware-io doesn't update values after validation

### ISSUE THREAD

**Issue by knopki** [marblejs/marble#131]
**Describe the bug**
Sometimes, I need to validate some data encoded as a sting with help of `io-ts`. For example, I have a query parameter `page` (`?page=123`)  that must be a number. So, I use custom type:
```js
const numberFromStringCodec = new t.Type<number, string, unknown>(
  'NumberFromStringCodec',
  t.number.is,
  (u: unknown, c: t.Context) => {
    const validation = t.string.validate(u, c);
    if (validation.isLeft() || u === '') {
      return t.failure(u, c);
    }
    const s = validation.value;
    const n = Number(s);
    return isNaN(n) ? t.failure(s, c) : t.success(n);
  },
  (n: number) => n.toString()
);
const data = { page: '1' };
const schema = t.interface({ page: numberFromStringCodec });
schema.decode(data) // right({ page: 1 }) <-- string changed to number
```
But `middleware-io` doesn't update validated values. So, after (successful) validation with `requestValidator$` I have new static type but old unchanged value.

**To Reproduce**
```js
...
r.useEffect((req$, _, { ask }) =>
  req$.pipe(
    use(
      requestValidator$({
        query: t.interface({ page: numberFromStringCodec })
      })
    ),
    mergeMap(async ({ req, state }) => {
      // `req.query.page`:
      //   static type is `number`
      //   runtime type is `string`
      //   value is '1' (should be 1)
    })
...
```
and send request with `?page=1`

**Expected behavior**
Same values as with manual run of `codec.decode()`.

**Desktop (please complete the following information):**
 - OS Linux
 - Package + Version @marblejs/core 2.1.1 + @marblejs/middleware-io 2.1.1
 - Node version 10.15.3

**Comment by JozefFlakus** [comment]
@knopki thanks for detecting this issue. The fix is on the way and will be included in the upcoming `v.2.2` release 💪

**Comment by JozefFlakus** [comment]
Fixed in [v2.2.0](https://docs.marblejs.com/other/changelog#2-2-0-2019-05-26)

**Comment by knopki** [comment]
Works great! Thanks so much.
